protocol/0078-NWLI-network_wide_limits.md

```diff
@@ -0,0 +1,80 @@
+
+The limit does not apply to liquidity providers.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.
+If it gets decreased below the current total party count in any given market then all parties in that market remain unaffected (their positions and orders remain open), but no new parties are allowed to participate in the market until the total party count drops below the new limit.
+
+## Limit orders on a book
+
+Introduce a new network parameter `limits.markets.maxLimitOrders` controlling the maximum number of limit orders that can rest on a book in any market. If the limit is reached limit orders can still be submitted, but:
+
+* if the order is aggressive (results in a trade) the trade proceeds as normal,
+* if the order is passive (it doesn't match on entry) and would get added to the order book it gets rejected.
+
+Pegged orders and [liquidity provision orders](./0038-OLIQ-liquidity_provision_order_type.md) do not get counted towards the limit.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.
+If it gets decreased below the current total number of orders on the book in any given market then all open orders in that market remain unaffected, but no new orders are allowed until the **total open order count** drops below the new limit.
+
+## Pegged orders on a market
+
+Introduce a new network parameter `limits.markets.maxPeggedOrders` controlling the maximum number of pegged orders that can rest on a book in any market.
+
+### Change of network parameter
+
+If the network parameter gets increased via a governance vote no further actions are needed.
+If it gets decreased below the current total number of orders on the book in any given market then all pegged orders already present in that market remain unaffected, but no new pegged orders are allowed until the **total pegged order count** drops below the new limit.
+
+[Liquidity provision orders](./0038-OLIQ-liquidity_provision_order_type.md) do not get counted towards the limit.
+
+## LP order shapes
+
+Each [LP order shape](./0038-OLIQ-liquidity_provision_order_type.md#how-they-are-submitted) has a hardcoded limit of at most 20 entries (offsets).
+
+## LPs in a market
+
+Introduce a new network parameter `limits.markets.maxLPs` controlling the maximum number of liquidity providers that can be active in any market. Each liquidity provider that successfully submits a liquidity provision transaction for a given market gets counted towards the **active LP count** for that market. When an LP in a given market reduces their commitment amount to 0 or gets closed out the LP count for that market gets decremented by 1.
+
+Once the active LP count in a market reaches the limit no further LP commitments are accepted. Once the count drops below the limit the LP commitments are accepted again.
```

Comment:
I think we will need to complicate this design a bit more to not allow a possibility of a market becoming deadlocked in a liquidity auction. 

So if market is in liquidity auction and the max LP limit is reached we should probably allow new LPs to kick out (their commitment gets cancelled) existing ones if their commitment is larger (in the commitment amount & seniority priority - we traverse the list by lowest commitment and within any commitment amount by time in market).

Reply:
Added now.